src/Slist.hs

```diff
@@ -672,6 +673,21 @@ init sl@Slist{..} = case sSize of
     _        -> Slist (P.init sList) (sSize - 1)
 {-# INLINE init #-}
 
+{- | @O(1)@. 'cons' is 'Slist' analogue to ':' for lists.
+It adds the given element to the beginning of the list.
```

Comment:
I think, in the documentation you can write that this function satisfies the following property:

```haskell
size (cons x xs) = size xs + 1
```

Since `size` returns `Size`, this property holds even for infinite lists 🙂